.github/workflows/e2e.yml

```diff
@@ -0,0 +1,130 @@
+
+    - name: CUSTOM KIND
+      run: |
+        wget -O kind https://github.com/painhardcore/builder/releases/download/0.0.1-alpha/kind_v0.8.1
+        sudo chmod +x kind
+        sudo mv kind /usr/bin
+          
+    - name: Kubectl install
+      run: |
+        curl -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl
+        chmod +x ./kubectl
+        sudo mv ./kubectl /usr/local/bin/kubectl
+        kubectl version --client 
+
+    - name: add CSI_VERSION
+      run: |
+        cd ${{env.CSI_BAREMETAL_DIR}}
+        echo "CSI_VERSION=`make version`" >> $GITHUB_ENV
+
+    - name: Get dependencies
+      run: make dependency
+
+    - name: Compile proto files
+      run: |
+        sudo apt install protobuf-compiler
+        sudo apt install golang-goprotobuf-dev
+        make install-compile-proto
+
+    - name: Generate CRD
+      run: |
+        make install-controller-gen 
+        make generate-deepcopy
+
+    - name: Build binary
+      run: |
+        make build
+        make DRIVE_MANAGER_TYPE=loopbackmgr build-drivemgr
+
+    - name: Download health-probe
+      run: make download-grpc-health-probe
+
+    - name: Build docker images
+      run: |
+        make images REGISTRY=${{ env.REGISTRY }}
+        make DRIVE_MANAGER_TYPE=loopbackmgr image-drivemgr REGISTRY=${{ env.REGISTRY }}
+    
+    - name: docker-build operators 
+      run: |
+        cd ${{env.CSI_BAREMETAL_OPERATOR_DIR}}
+        echo "CSI_OPERATOR_VERSION=`make version`" >> $GITHUB_ENV
+        make docker-build REGISTRY=${{ env.REGISTRY }}
+
+    - name: Kind preparation
+      run: |
+        cd ${{env.CSI_BAREMETAL_DIR}}
+        make kind-build
+        make kind-create-cluster
+        kubectl cluster-info --context kind-kind
+        kubectl get pods -o wide --all-namespaces 
+        echo "current-context:" $(kubectl config current-context)
+        echo "environment-kubeconfig:" ${KUBECONFIG}
+        sudo sysctl net/netfilter/nf_conntrack_max=131072 
+      
+    - name: Sleep for 60 seconds
+      uses: jakejarvis/wait-action@master
+      with:
+        time: '60s' 
+
+    - name: Check pods
+      run: |
+        kubectl cluster-info --context kind-kind
+        kubectl get pods -o wide --all-namespaces 
+
+    - name: Sidecars preparing
+      run: |
+        make deps-docker-pull
+        make deps-docker-tag
+
+    - name: Retag CSI images and load them to kind
+      run: |
+        docker images 
+        make kind-tag-images TAG=$CSI_VERSION REGISTRY=${{ env.REGISTRY }}
+        make kind-load-images TAG=$CSI_VERSION REGISTRY=${{ env.REGISTRY }}
+        make kind-tag-load-operator-image OPERATOR_VERSION=$CSI_OPERATOR_VERSION REGISTRY=${{ env.REGISTRY }}
+          
+    - name: Make test
+      run: make test-ci CSI_VERSION=$CSI_VERSION OPERATOR_VERSION=$CSI_OPERATOR_VERSION CHARTS_DIR=${{env.CSI_BAREMETAL_OPERATOR_DIR}}charts
```

Comment:
Could you please also add a step to load log and result files to artifacts. Log file should be log.txt, results -  test/e2e/report.xml. Reference - https://github.com/dell/csi-baremetal/blob/2a1958a294373eaf8450d2598abe26b1de6a33be/.github/workflows/pr.yml#L59